### PR TITLE
Bump vsphere-csi-driver to v2.4.2

### PR DESCRIPTION
**Pull request by abhay-krishna**
Bumping vsphere-csi-driver to the [latest release](https://github.com/kubernetes-sigs/vsphere-csi-driver/releases/tag/v2.6.0).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/kubernetes-sigs/vsphere-csi-driver/Makefile [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 GIT_TAG:=$(shell cat GIT_TAG)
-GOLANG_VERSION?="1.15"
+GOLANG_VERSION?="1.17"
 
 REPO=vsphere-csi-driver
 REPO_OWNER=kubernetes-sigs
 
 BINARY_TARGET_FILES=vsphere-csi-driver vsphere-csi-syncer
 SOURCE_PATTERNS=./cmd/vsphere-csi ./cmd/syncer
+BINARY_TARGET_FILES_BUILD_ALONE=vsphere-csi-syncer
```
Comment (abhay-krishna) [inline projects/kubernetes-sigs/vsphere-csi-driver/Makefile:10]: @jaxesn We weren't setting the [version LDFLAGS](https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/v2.6.0/Makefile#L91-L92) for these Go builds, and they differ for the driver and syncer binaries, so building this separately.

**Comment by jaxesn** [comment]
Actually now that I think about this, I'm not sure you can do this. We need to stay on a version which supports 1.20 until we break out into release branch builds

/hold

**Comment by abhay-krishna** [comment]
> Actually now that I think about this, I'm not sure you can do this. We need to stay on a version which supports 1.20 until we break out into release branch builds
> 
> /hold

Downgraded to v2.4.2 which supports 1.20 through 1.22 according to the VMware docs.

File: projects/kubernetes-sigs/vsphere-csi-driver/Makefile [hunk 2]
```diff
@@ -16,19 +18,11 @@ CSI_SYNCER_IMAGE_COMPONENT=kubernetes-sigs/vsphere-csi-driver/csi/syncer
...
 include $(BASE_DIRECTORY)/Common.mk
 
-csi-driver/images/%: BASE_IMAGE_NAME=eks-distro-minimal-base-csi
-
-$(call IMAGE_TARGETS_FOR_NAME, csi-syncer): $(OUTPUT_FILES_TARGET)
+$(OUTPUT_BIN_DIR)/linux-%/vsphere-csi-syncer: EXTRA_GO_LDFLAGS=-X sigs.k8s.io/vsphere-csi-driver/v2/pkg/syncer.Version=$(GIT_TAG)
 
-$(OUTPUT_FILES_TARGET):
-	@mkdir -p $@
-	cp $(REPO)/pkg/apis/cnsoperator/config/cnsregistervolume_crd.yaml \
```
[the remark this answers is not in the document]
Comment (abhay-krishna) [reply projects/kubernetes-sigs/vsphere-csi-driver/Makefile:29]: They removed these files from the Dockerfile. They were being copied in using `ADD` instructions before.

**Review by jaxesn — APPROVED** [review]

**Pull request merged**